src/controllers/things_controller.js

```diff
@@ -34,22 +35,60 @@ ThingsController.get('/', function (request, response) {
 /**
  * Handle creating a new thing.
  */
-ThingsController.post('/', function (request, response) {
+ThingsController.post('/', async (request, response) => {
   if (!request.body || !request.body.id) {
     response.status(400).send('No id in thing description');
     return;
   }
-  var description = request.body;
-  var id = description.id;
+  const description = request.body;
+  const id = description.id;
   delete description.id;
-  Things.createThing(id, description).then(function(thing) {
+
+  // If we're adding a native webthing, we need to update the config for
+  // thing-url-adapter so that it knows about it.
+  let webthing = false;
+  if (description.hasOwnProperty('webthingUrl')) {
+    webthing = true;
+
+    const key = 'addons.thing-url-adapter';
+    try {
+      const current = await Settings.get(key);
+      if (typeof current === 'undefined') {
+        throw new Error('Setting is undefined.');
+      }
+
+      current.moziot.config.urls.push(description.webthingUrl);
```

Comment:
What happens if the list of urls already contains the web thing's url? (user presses enter twice or just forgets)

Reply:
It will be handled by the adapter just fine, since the adapter keeps track of "known URLs" when loading things.